### PR TITLE
Drop EL7 support

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by jhoblitt** [comment]
Please don't drop EL7 yet... I'm still working on migrating my production foreman deployments to EL8.

**Comment by ehelms** [comment]
> Please don't drop EL7 yet... I'm still working on migrating my production foreman deployments to EL8

Can you share your timeline? We've been communicating this drop [1] and timeline [2].

[1] https://community.theforeman.org/t/deprecation-plans-for-foreman-on-el7-debian-10-and-ubuntu-18-04/25008
[2] https://community.theforeman.org/t/the-el7-deprecation-checklist/28041/4

A release of this module will exist with continued EL7 support. This drop will affect the main branch of this module and the nightly Foreman ecosystem. As we remove EL7 nightly repositories this module will be broken if we do not drop the EL7 support from the module itself.

**Comment by alexjfisher** [comment]
@ehelms Any chance of getting a release out that includes @jhoblitt 's https://github.com/theforeman/puppet-foreman/pull/1054 before EL7 support is removed?

**Comment by ehelms** [comment]
> @ehelms Any chance of getting a release out that includes @jhoblitt 's #1054 before EL7 support is removed?

Aye, we can do that.

**Comment by ehelms** [comment]
@alexjfisher @jhoblitt the requested release is available and we can cut future ones if needed https://forge.puppet.com/modules/theforeman/foreman/20.2.0

File: manifests/repos.pp [hunk 1]
```diff
@@ -8,7 +8,6 @@
   case $facts['os']['family'] {
     'RedHat', 'Linux': {
```
Comment (ekohl) [inline manifests/repos.pp:9]:
```suggestion
    'RedHat': {
```

File: manifests/repos.pp [hunk 2]
```diff
@@ -8,7 +8,6 @@
...
       $yumcode = $facts['os']['name'] ? {
-        'Amazon' => 'el7',
         'Fedora' => "f${facts['os']['release']['major']}",
         default  => "el${facts['os']['release']['major']}",
       }
```
Comment (ekohl) [inline manifests/repos.pp:13]: We haven't had Fedora releases for a long time. Simplify it to `$yumcode = "el${facts['os']['release']['major']}"` perhaps? Or even inline it in the `foreman::repos::yum` declaration a few lines below.

File: spec/spec_helper_acceptance.rb [hunk 3]
```diff
@@ -12,13 +12,6 @@
     # refresh check if cache needs refresh on next yum command
     on host, 'yum clean expire-cache'
   end
-
-  # In Puppet 7 the locale ends up being C.UTF-8 if it isn't passed.
```
Comment (ekohl) [inline spec/spec_helper_acceptance.rb:16]: Please also drop this bit:
https://github.com/theforeman/puppet-foreman/blob/a71cb9e4b5b26549efe0f78f941ceaca00e99b20/.sync.yml#L10-L11=

**Review by ekohl — CHANGES_REQUESTED** [review]
I only looked at the actual code and currently I'm not answering the "should we drop it now or keep support in the module for a while" question.

**Comment by ehelms** [comment]
Shall we take in this change given EL7 has been dropped from nightly and thus will be for Foreman 3.4?

**Review by ekohl — COMMENTED** [review]
If you also modify https://github.com/theforeman/puppet-foreman#foreman-version-compatibility-notes I think this can be merged.

**Comment by ehelms** [comment]
Let me know thoughts on that wording.

**Review by ekohl — APPROVED** [review]

**Pull request merged**